sdk/programmableconnectivity/azure-programmableconnectivity/samples/number_verification_sample.py

```diff
@@ -0,0 +1,42 @@
+## ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+from azure.programmableconnectivity import ProgrammableConnectivityClient
+from azure.identity import DefaultAzureCredential
+from azure.core.exceptions import HttpResponseError
+from azure.programmableconnectivity.models import (
+    NetworkIdentifier,
+    NumberVerificationWithCodeContent,
+    NumberVerificationWithoutCodeContent,
+)
+
+# This flow involves 2 calls, with the user of the SDK needing to follow the redirect URI given in the `location` header value from the first call.
+
+client = ProgrammableConnectivityClient(endpoint="<endpoint>", credential=DefaultAzureCredential())
+APC_GATEWAY_ID = "/subscriptions/<subscription_id>/resourceGroups/.../.../..."
+
+location = None
+
+
+def callback(response):
+    global location
+    location = response.http_response.headers.get("location")
+
+
+network_identifier = NetworkIdentifier(identifier_type="NetworkCode", identifier="Telefonica_Brazil")
+
+content = NumberVerificationWithoutCodeContent(
+    phone_number="<number>", redirect_uri="<redirect_uri>", network_identifier=network_identifier
+)
+client.number_verification.verify_without_code(body=content, apc_gateway_id=APC_GATEWAY_ID, raw_response_hook=callback)
```

Comment:
@xiangyan99 This code doesn't work, and I'm working on fixing it. The issue is that `verify_without_code` gets back a 302, and we need the redirect to NOT be followed, and to just get the `location` header back.

I thought I could solve this by having the code

```python
from azure.core.pipeline.policies import RedirectPolicy
redirect_policy = RedirectPolicy.no_redirects()
client = ProgrammableConnectivityClient(
    endpoint="https://eastus.prod.apcgatewayapi.azure.com",
    credential=credential,
    redirect_policy=redirect_policy,
)
```

Which would then disable the redirections, as this would get picked up in `ProgrammableConnectivityClientConfiguration`. However my testing shows that this isn't the case, and the request still hangs because it's trying to follow the `302`, and then hangs.

Could you offer me any pointers to what else could be going wrong?

Reply:
Update: Tried to debug this today, for some reason it's just working though.

Reply:
So running the code with `python number_verification_sample.py` works, but as soon as I put it into tests, it hangs.

```
class TestAzureProgrammableConnectivity(AzureRecordedTestCase):
...
    @recorded_by_proxy
    def test_number_verification_without_code(self, **kwargs):
        <same code as in snippet>
```

@xiangyan99 does it make sense for this to happen? I can imagine that having a proxy makes redirects slightly different.

Reply:
I'm taking a look at this and will attempt to reproduce and fix the issue (Aron has a thread open in the test proxy questions Teams channel). The redirect following is likely coming from the test proxy tool itself, which explains why things hang when the test proxy is invoked, but not when live tests are run with `AZURE_SKIP_LIVE_RECORDING=true` (which bypasses the proxy entirely).

Updating recording options to disable redirects didn't seem to work, so I'm investigating why that is. If we can't sort it out in the short term, a temporary solution would be to make this test live-only so it can still run but doesn't invoke the test proxy.

Reply:
@mccoyp any update on the 302s?

Reply:
@aronhegedusms thanks for the ping -- I merged a fix for redirect settings today: https://github.com/Azure/azure-sdk-for-python/pull/35510

If you rebase your branch with the latest from `main`, calling `set_function_recording_options(handle_redirects=False)` at the start of your test should fix the issue. I tested on my end with a flask app since I couldn't get Programmable Connectivity resources set up to test; I know that the linked PR makes the test proxy work as expected, and I'm pretty sure it'll fix the problem you're seeing.

Reply:
Perfect, rebased, and worked as expected. Thank you for your help in fixing this issue.